src/System.Devices.Gpio/GPIORef.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Devices.Gpio
+{
+    /// <summary>
+    /// GPIO - Basic Scenarios
+    /// 
+    /// Most GPIO implementations share a core set of functionality to allow basic on/off control
+    /// of pins. At the very least, we should support these.
+    /// - Open a object that is a representation of a pin with the given pin number
+    /// - Support closing a pin to release the resources owned by that pin object
+    /// - Represent the mode of the pin that details how the pin handles reads and writes (e.g. Input, Output)
+    /// - Allow a resistor to be added to a pin such that it can be set as pullup or pulldown (or no pull)
+    /// - Support setting a PWM value on a pin
+    /// </summary>
+    #region GPIO - Basic Scenarios
+
+    public partial class GPIOPin : IDisposable
+    {
+        int PinNumber { get { throw new NotImplementedException(); } }
+        bool Read() { throw new NotImplementedException(); }
+        void Write(bool value) { throw new NotImplementedException(); }
+        void Dispose() { throw new NotImplementedException(); }
+        GPIOPin(GPIOController controller, int pinNumber, GPIOPinMode pinMode = GPIOPinMode.Input) { }
+
+        GPIOPinMode PinMode { get { throw new NotImplementedException(); } set { throw new NotImplementedException(); } }
+        bool IsPinModeSupported(GPIOPinMode pinMode) { throw new NotImplementedException(); }
+        int PWMValue { set { throw new NotImplementedException(); } }
```

Comment:
I searched for PVM and nothing interesting came up in Bing and Google search. This means we should not use the acronym. :-)

Reply:
[Pulse-width_modulation](https://www.bing.com/search?q=pwm&qs=n&form=QBRE&sp=-1&pq=pwm&sc=8-3&sk=&cvid=6AB8F0887B0E4283B00ADFE3A6D8B683)